### PR TITLE
PHPUnit tests: Adminhtml/Order/CreateController

### PR DESCRIPTION
**Pull request by YevhenShBolt**
Task - https://app.asana.com/0/941895179897714/921469870263919/f

### CODE REVIEW

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                'getRequest', '_redirect', '_processActionData', '_getSession',
+                'getImmutableQuoteIdFromTransaction', '_normalizeOrderData',
+                '_getOrderCreateModel', '_postOrderCreateProcessing'
+            ])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $session = new Mage_Adminhtml_Model_Session_Quote();
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', true);
```
Comment (daisy1754) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: let's use realistic data in test - I think 'bolt_reference' is a string with form 'AAAA-BBBB-1234'

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    public function testSaveActionWhereBoltReferenceIsFalse()
+    {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest', 'getResponse', '_redirect'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', false);
```
Comment (daisy1754) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: I think in the real world, bolt_reference is undefined, not false

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            ->setMethods(['getRequest'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $result = $this->currentMock->prepareAddressData($postData);
+
+        $this->assertEquals($actual, $result);
+    }
+
+    public function testSaveActionWhereBoltReferenceIsFalse()
```
Comment (daisy1754) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: can you add comment what exactly you test (eg., "if Bolt reference is undefined, we don't create order")

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    /**
+     * @var
+     */
+    private $currentMock;
+
+    public function setUp()
+    {
+    }
+
+    public function testLoadBlockAction()
```
Comment (daisy1754) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: testLoadBlockAction_returnsAddressFromPrepareAddressData

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+        $this->currentMock->method('getResponse')
+            ->willReturn($response);
+
+        $this->currentMock->method('_redirect')
+            ->willReturn(false);
+
+        $result = $this->currentMock->saveAction();
+
+        $this->assertNull($result);
```
Comment (daisy1754) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: I'm not sure if this is the assertion we want. For successful save order case this method returns null too, right?

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+        $this->currentMock->method('getRequest')
+            ->willReturn($request);
+
+        $this->currentMock->method('_normalizeOrderData')
+            ->willReturn(false);
+        $this->currentMock->method('_processActionData')
+            ->willReturn(true);
+
+        $this->currentMock->method('_redirect')
```
Comment (daisy1754) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: can we also check redirect url?

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php (hunk 5 above)
Comment (YevhenShBolt) [reply tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: If bolt_reference is false or not exist the original controller will return nothing, the code below:

```   
   $boltReference = $this->getRequest()->getPost('bolt_reference');
   if (!$boltReference) {
        $this->_normalizeOrderData();  // We must re-normalize the data first
        parent::saveAction();
        return;
   }
```

But i am not sure if `parent::saveAction();` will save anything.

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php (hunk 1 above)
Comment (YevhenShBolt) [reply tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: fixed.

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php (hunk 2 above)
Comment (YevhenShBolt) [reply tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: fixed.

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php (hunk 6 above)
Comment (YevhenShBolt) [reply tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: I have no idea how to check redirect. Can you give an advice?

Comment (daisy1754) [reply tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: can you capture the argument of method call with phpunit?

File: tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php (hunk 5 above)
Comment (daisy1754) [reply tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php]: parent::saveAction will create order in database, right? Can we check if order is created in db?

**Comment by YevhenShBolt** [comment]
As discussed with Kazuki i will add a proper boolean return to the class.

**Comment by matt-thomason** [comment]
@daisy1754 dismissing review since it is resolved and changes have been addressed.

**Pull request merged**